app/src/main/kotlin/com/simplemobiletools/gallery/pro/activities/MainActivity.kt

```diff
@@ -345,20 +346,22 @@ class MainActivity : SimpleActivity(), DirectoryOperationsListener {
 
     private fun refreshMenuItems() {
         if (!mIsThirdPartyIntent) {
-            main_menu.getToolbar().menu.apply {
+            binding.mainMenu.getToolbar().menu.apply {
                 findItem(R.id.column_count).isVisible = config.viewTypeFolders == VIEW_TYPE_GRID
                 findItem(R.id.set_as_default_folder).isVisible = !config.defaultFolder.isEmpty()
                 findItem(R.id.open_recycle_bin).isVisible = config.useRecycleBin && !config.showRecycleBinAtFolders
-                findItem(R.id.more_apps_from_us).isVisible = !resources.getBoolean(R.bool.hide_google_relations)
+                findItem(R.id.more_apps_from_us).isVisible = !resources.getBoolean(com.simplemobiletools.commons.R.bool.hide_google_relations)
             }
         }
 
-        main_menu.getToolbar().menu.apply {
+        binding.mainMenu.getToolbar().menu.apply {
             findItem(R.id.temporarily_show_hidden).isVisible = !config.shouldShowHidden
             findItem(R.id.stop_showing_hidden).isVisible = (!isRPlus() || isExternalStorageManager()) && config.temporarilyShowHidden
 
             findItem(R.id.temporarily_show_excluded).isVisible = !config.temporarilyShowExcluded
             findItem(R.id.stop_showing_excluded).isVisible = config.temporarilyShowExcluded
+
+            findItem(R.id.access_more_media).isVisible = isUpsideDownCakePlus()
```

Comment:
this should check some permissions too, not just the OS version. Like the button is useless for example at upgrading from older app versions as everything needed is granted already. Also if the user grants access to all media files or all files in general, this menu item shouldnt be visible.

Reply:
that file picker system doesnt make sense in the gallery actually, it was supposed to be used in other apps. And it doesnt work either... If I pick some images the app can access, it still closes and shows the Storage permission is required toast. But runs well on next try.

Reply:
I have removed it now. Now the app will always display that toast unless user selects `Allow all`.